### PR TITLE
Add `period` field to `StatsRecord`

### PR DESCRIPTION
**Pull request by jklausa**
I started doodling around on connecting `Core Data` to our `Store`s and quickly realized I forgot a _fairly important field when modeling the Core Data entities 🤦‍♂️ . This fixes it :)

To test:
Verify the tests still pass

### CODE REVIEW

File: WordPress/Classes/Utility/PushNotificationsManager.swift [hunk 1]
```diff
@@ -444,4 +444,3 @@ private extension Date {
                                                from: self)
     }
 }
-
```
Comment (jklausa) [inline WordPress/Classes/Utility/PushNotificationsManager.swift]: linter yelled at me about this line, even though I haven't changed anything here???? so i just removed the whitespace to appease it.

**Comment by jklausa** [comment]
Hmm, those tests didn't fail locally... Marking it as DNR just yet.

**Comment by jklausa** [comment]
Okay, I think this should be it now.

**Comment by jklausa** [comment]
oh yikes this did not work as planned

**Comment by jklausa** [comment]
(I tried merging `develop` into this branch and It Did Not Go Well. 
I: 
1. merged `develop` into `feature/stats-core-data-finished` 
2. pushed `feature/stats-core-data-finished` 
2. merged `feature/stats-core-data-finished` into this branch

to fix it. 

I think (?????) it's fine for me to just merge `develop` into `feature/stats-core-data-finished` without going thru PR process, but feel free to yell at me if you disagree ;P

File: WordPress/Classes/Utility/PushNotificationsManager.swift (hunk 1 above)
Comment (ScoutHarris) [reply WordPress/Classes/Utility/PushNotificationsManager.swift]: You merged `develop` didn't you? 😄 (I had the same problem.)

**Comment by ScoutHarris** [comment]
> I think (?????) it's fine for me to just merge `develop` into `feature/stats-core-data-finished` without going thru PR process

If you mean merge `feature/stats-core-data-finished` into `develop`, I think that would be fine. If that's not what you meant, I'm confused. 😄

**Comment by jklausa** [comment]
No, the other way around ;P

I meant that I merged `develop` back into the `feature/stats-core-data-finished` feature branch locally and pushed it, without submitting a whole PR for it — I think it's not a breach of our procedures, but I'm actually not sure ;P

**Comment by ScoutHarris** [comment]
Oh I see! Naw, I think that's fine. 👍

**Pull request merged**